File: src/main/scala/com/lightbend/akka/sample/LogAllTheThingsActor.scala
/*
 * Copyright (C) 2017 Lightbend Inc. <http://www.typesafe.com>
 */
package com.lightbend.akka.sample

import akka.actor.{Actor, ActorLogging, Props}


object LogAllTheThingsActor {
  def props(messageTransformer: Option[PartialFunction[Any, String]] = None) =
    Props(new LogAllTheThingsActor(messageTransformer))

}

class LogAllTheThingsActor(messageTransformer: Option[PartialFunction[Any, String]]) extends Actor with ActorLogging {
  override def receive: Receive = {
    case message =>
      val logEntry = messageTransformer.flatMap(transformer =>
        transformer.lift(message)
      ).getOrElse(s"Saw message: [$message]")

      log.info(logEntry)
  }
}

File: src/main/java/com/ligthbend/akka/cluster/java/sample/LogAllTheThingsActor.java
/**
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */
package com.ligthbend.akka.cluster.java.sample;

import akka.actor.AbstractLoggingActor;
import akka.actor.Props;

public final class LogAllTheThingsActor extends AbstractLoggingActor {

  public static Props props() {
    return Props.create(LogAllTheThingsActor.class, LogAllTheThingsActor::new);
  }

  @Override
  public Receive createReceive() {
    return receiveBuilder()
        .match(Object.class, message -> log().info("Saw message {}", message))
        .build();
  }
}
